rholang/src/test/scala/coop/rchain/rholang/interpreter/error/MLockSpec.scala

```diff
@@ -0,0 +1,62 @@
+package coop.rchain.rholang.interpreter.error
+
+import cats.effect.concurrent.Ref
+import cats.implicits._
+import monix.eval.Task
+import org.scalatest.{FlatSpec, Matchers}
+
+import scala.concurrent.duration._
+
+class MLockSpec extends FlatSpec with Matchers {
+
+  import monix.execution.Scheduler.Implicits.global
+
+  private val MAX_DURATION = 10.seconds
+
+  def withFixture[A](f: MLock[Task] => Task[A]): A =
+    (MLock[Task] >>= f).runSyncUnsafe(MAX_DURATION)
+
+  behavior of "MLock"
+
+  "release" should "throw error when released lock is released" in withFixture(
+    _.release.attempt
+      .map(_ should be(Left(LockReleaseException("attempted to release released lock"))))
+  )
+
+  "release" should "unblock lock acquisitions in FIFO order" in withFixture { lock =>
+    for {
+      ref <- Ref.of[Task, Int](0)
+      _   <- lock.acquire
+      result <- List(
+                 lock.acquire >> ref.getAndSet(1),
+                 Task.sleep(0.1.seconds) >> lock.acquire >> ref.getAndSet(2),
```

Comment:
Using `sleep` can lead to potential race conditions and test failures in rare cases. Consider using this approach instead: https://monix.io/docs/3x/execution/scheduler.html#injecting-time-and-tests